log4j-core/src/main/java/org/apache/logging/log4j/core/appender/nosql/NoSqlConnection.java

```diff
@@ -49,8 +51,21 @@ public interface NoSqlConnection<W, T extends NoSqlObject<W>> extends Closeable
      * Inserts the given object into the underlying NoSQL database.
      *
      * @param object The object to insert.
+     * @deprecated see {@link #insertObject(LogEvent, NoSqlObject)}
+     */
+    @Deprecated
```

Comment:
I don't see the pressing need to deprecate this method. It will generate compiler warnings on legit user code implementing this method.
```suggestion
```